### PR TITLE
SOLR-15225: maintain existing test naming conventions (contrib/analytics, contrib/ltr)

### PR DESCRIPTION
**Pull request by cpoerschke**
https://issues.apache.org/jira/browse/SOLR-15225

This pull request shows how different localised existing test naming conventions could be maintained temporarily whilst we figure out what convention or conventions we want overall for Solr permanently.

### CODE REVIEW

File: solr/test-framework/src/java/org/apache/solr/SolrTestCase.java [hunk 1]
```diff
@@ -19,10 +19,12 @@
 
 import java.lang.invoke.MethodHandles;
 import java.io.File;
+import java.util.regex.Pattern;
 
 import com.carrotsearch.randomizedtesting.annotations.ThreadLeakLingering;
 import org.apache.lucene.util.LuceneTestCase;
 import org.apache.lucene.util.QuickPatchThreadsFilter;
+import org.apache.lucene.util.VerifyTestClassNamingConvention;
```
Comment (cpoerschke) [inline solr/test-framework/src/java/org/apache/solr/SolrTestCase.java:27]: Hmm, how mysterious, https://github.com/apache/lucene/blob/main/lucene/test-framework/src/java/org/apache/lucene/util/QuickPatchThreadsFilter.java and https://github.com/apache/lucene/blob/main/lucene/test-framework/src/java/org/apache/lucene/util/VerifyTestClassNamingConvention.java are both "neighbours" but somehow there's difficulty discovering `VerifyTestClassNamingConvention` somehow? Had seen this locally but assumed it must be my setup somehow, hmm.

Comment (cpoerschke) [reply solr/test-framework/src/java/org/apache/solr/SolrTestCase.java:27]: Merging in the latest `origin/main` changes somehow solved this, yay!

**Review by dweiss — APPROVED** [review]
Sorry, Christine - didn't see the request. It's fine with me. I think with time it should be made consistent though, right? It's a temporary measure to have two different conventions?

**Review by HoustonPutman — APPROVED** [review]
Looks good to me for the meantime.

**Review by alessandrobenedetti — APPROVED** [review]
It seems good to me!

**Comment by cpoerschke** [comment]
> ... I think with time it should be made consistent though, right? It's a temporary measure to have two different conventions?

Yes and yes, I think so.

**Pull request merged**